apps/android/app/src/main/java/chat/simplex/app/views/chat/item/CIVoiceView.kt

```diff
@@ -0,0 +1,344 @@
+      val metaReserve = if (edited)
+        "                     "
+      else
+        "                 "
+      Text(metaReserve)
+    }
+  }
+}
+
+@Composable
+private fun fileIcon(
+  innerIcon: ImageVector? = null,
+  color: Color = if (isInDarkTheme()) FileDark else FileLight
+) {
+  Box(
+    contentAlignment = Alignment.Center
+  ) {
+    Icon(
+      Icons.Filled.InsertDriveFile,
+      stringResource(R.string.icon_descr_file),
+      Modifier.fillMaxSize(),
+      tint = color
+    )
+    if (innerIcon != null) {
+      Icon(
+        innerIcon,
+        stringResource(R.string.icon_descr_file),
+        Modifier
+          .size(32.dp)
+          .padding(top = 12.dp),
+        tint = Color.White
+      )
+    }
+  }
+}
+
+private fun fileAction(
+  file: CIFile?,
+  receiveFile: (Long) -> Unit,
+  saveFileLauncher: ManagedActivityResultLauncher<String, Uri?>
+) {
+  if (file != null) {
+    when (file.fileStatus) {
+      CIFileStatus.RcvInvitation -> {
+        if (fileSizeValid(file)) {
+          receiveFile(file.fileId)
+        } else {
+          AlertManager.shared.showAlertMsg(
+            generalGetString(R.string.large_file),
+            String.format(generalGetString(R.string.contact_sent_large_file), formatBytes(MAX_FILE_SIZE))
+          )
+        }
+      }
+      CIFileStatus.RcvAccepted ->
+        AlertManager.shared.showAlertMsg(
+          generalGetString(R.string.waiting_for_file),
+          String.format(generalGetString(R.string.file_will_be_received_when_contact_is_online), MAX_FILE_SIZE)
+        )
+      CIFileStatus.RcvComplete -> {
+        val filePath = getLoadedFilePath(SimplexApp.context, file)
+        if (filePath != null) {
+          saveFileLauncher.launch(file.fileName)
+        } else {
+          Toast.makeText(SimplexApp.context, generalGetString(R.string.file_not_found), Toast.LENGTH_SHORT).show()
+        }
+      }
+      else -> {}
+    }
+  }
+}
+
+@Composable
+private fun fileIndicator(
+  file: CIFile?,
+  audioPlaying: Boolean,
+  sent: Boolean,
+  hasText: Boolean,
+  audioInfo: State<ProgressAndDuration>?,
+  receiveFile: (Long) -> Unit,
+  play: () -> Unit,
+  pause: () -> Unit
+) {
+  if (file != null && file.loaded && audioInfo != null) {
+    val strokeWidth = with(LocalDensity.current){ 2.dp.toPx() }
+    val primary = MaterialTheme.colors.primary
+    val angle = remember { derivedStateOf { 360f * (audioInfo.value.progressMs.toDouble() / audioInfo.value.durationMs).toFloat() } }
+    if (hasText) {
+      IconButton({ if (!audioPlaying) play() else pause() }, drawRingModifier(angle.value, primary, strokeWidth)) {
+        Icon(
+          imageVector = if (audioPlaying) Icons.Filled.Pause else Icons.Filled.PlayArrow,
+          contentDescription = null,
+          Modifier.size(36.dp),
+          tint = MaterialTheme.colors.primary
+        )
+      }
+    } else {
+      Box(drawRingModifier(angle.value, primary, strokeWidth)) {
+        FloatingActionButton(
```

Comment:
I don't think it's a good idea to use `FloatingActionButton` for this - it has many things we don't need/use, and semantically it's not a FAB, just a message icon. Reading FAB's code it seems what we need from it is mostly that it's `Surface` with `MaterialTheme.shapes.small.copy(CornerSize(percent = 50))` Shape, maybe just use `Surface` and add what's necessary?

Reply:
What makes you thinking that we need to, basically, recreate a FAB code ourselves? FAB is just surface, yes, so you suggest to use Surface... with all setup that FAB already do

Reply:
What we will win here if we choose Surface?